### DISCUSSION TITLE
An instance wasn't initiated yet error

### DISCUSSION THREAD

**Discussion by Michael21107** [scripthookvdotnet/scripthookvdotnet#1375] [Q&A] [answered]
what does this mean?
![image](https://github.com/scripthookvdotnet/scripthookvdotnet/assets/121039033/26b13534-00fc-4da7-b51b-1931beb1f9fb)

**Comment by kagikn** [comment, marked as answer]
Not your fault, but the message "A script tried to use a custom script instance of type {} that was not instantiated by ScriptHookVDotNet." tells you the script used SHVDN incorrectly. The script tried to instantiate an instance that is a subclass of `GTA.Script` by their code, i.e. constructed a new instance by the `new` keyword (SHVDN doesn't register scripts instantiated by scripts' code). Maybe it instantiated a subclass of `GTA.Script` to communicate some data, but there's no need to do this for that purpose.

The lines that tells you that a script used `GTA.Script` incorrectly:
https://github.com/scripthookvdotnet/scripthookvdotnet/blob/893d92d47b8222a70e398dbc8cabc67afe55f12c/source/core/ScriptDomain.cs#L918-L924

**Reply by Michael21107** [reply]
thank you
